R/LongitudinalQuantities.R

```diff
@@ -0,0 +1,214 @@
+        time_grid = time_grid
+    )
+}
+
+
+
+
+#' `as.data.frame`
+#'
+#' @param x ([`LongitudinalQuantities`]) \cr Longitudinal Quantities
+#' @param ... Not used
+#' @family LongitudinalQuantities
+#' @export
+as.data.frame.LongitudinalQuantities <- function(x, ...) {
+    res <- as.data.frame(
+        x@quantities,
+        time_grid = x@time_grid,
+        groups = x@groups,
+        type = "na"
+    )
+    res["type"] <- NULL
+    return(res)
+}
+
+
+
+#' summary
+#'
+#'
+#' @description
+#' This method returns a `data.frame` of the longitudinal quantities
+#'
+#' @param object ([`LongitudinalQuantities`]) \cr Survival Quantities.
+#' @param conf.level (`numeric`) \cr confidence level of the interval.
+#' @param ... Not used.
+#'
+#' @family LongitudinalQuantities
+#' @family summary
+#' @export
+summary.LongitudinalQuantities <- function(
+    object,
+    conf.level = 0.95,
+    ...
+) {
+    res <- summary(
+        object@quantities,
+        time_grid = object@time_grid,
+        groups = object@groups,
+        type = "na",
+        conf.level = conf.level
+    )
+    res["type"] <- NULL
+    return(res)
+}
+
+
+#' Longitudinal Plot
+#'
+#' Internal plotting function to create longitudinal plots
+#' This function predominately exists to extract core logic into its own function
+#' to enable easier unit testing.
+#'
+#' @param data (`data.frame`)\cr A `data.frame` of summary statistics for longitudinal
+#' value estimates. See details.
+#' @param data_obs (`data.frame`)\cr A `data.frame` of real observed values to be
+#' overlaid for reference.  See details.
+#' @param add_ci (`logical`)\cr Should confidence intervals be added? Default = `TRUE`.
+#' @details
+#'
+#' ## `data`
+#' Should contain the following columns:
+#' - `time` - Time point
+#' - `group` - The group in which the observation belongs to
+#' - `median` - The median value for the summary statistic
+#' - `upper` - The upper 95% CI for the summary statistic
+#' - `lower` - The lower 95% CI for the summary statistic
+#'
+#' ## `data_obs`
+#' Should contain the following columns:
+#' - `time` - The time at which the observed value occurred
+#' - `Yob` - The real observed value
+#' - `group` - Which group the event belongs to, should correspond to values in `data$group`
+#' @keywords internal
+longitudinal_plot <- function(
+    data,
+    data_obs = NULL,
+    add_ci = FALSE
+) {
+    p <- ggplot() +
+        geom_line(aes(x = .data$time, y = .data$median), data = data) +
+        xlab(expression(t)) +
+        ylab(expression(y)) +
+        facet_wrap(~group) +
+        theme_bw()
+
+    if (add_ci) {
+        p <- p + geom_ribbon(
+            aes(x = .data$time, ymin = .data$lower, ymax = .data$upper),
+            data = data,
+            alpha = 0.3
+        )
+    }
+
+    if (!is.null(data_obs)) {
+        p <- p + geom_point(aes(x = .data$time, y = .data$Yob), data = data_obs)
+    }
+    return(p)
+}
+
+
+#' Automatic Plotting for LongitudinalQuantities
```

Comment:
```suggestion
#' Automatic Plotting for `LongitudinalQuantities`
```
would put as code any non-English stuff